homeassistant/components/unifiprotect/migrate.py

```diff
@@ -30,6 +35,32 @@ async def async_migrate_data(
     await async_migrate_device_ids(hass, entry, protect)
     _LOGGER.debug("Completed Migrate: async_migrate_device_ids")
 
+    entity_registry = er.async_get(hass)
+    created_issue = False
+    for entity in er.async_entries_for_config_entry(entity_registry, entry.entry_id):
+        if (
+            entity.domain == Platform.SENSOR
+            and entity.disabled_by is None
+            and "detected_object" in entity.unique_id
+            and (
+                automations_with_entity(hass, entity.entity_id)
+                or scripts_with_entity(hass, entity.entity_id)
```

Comment:
We should help/guide the user on which automations/scripts are involved.
(Similar to e.g., how we raise a repairs issue when we detect a call to a non-existing service).

Reply:
Well, as @epenet mentioned, `automations_with_entity` and `scripts_with_entity` are not full proof. And it cannot detect usages inside of templates. As a result, I do not think this issue should "fix itself" at all, but everyone else seems to think it should. I _definitely_ think telling the user an incomplete list of automations that are using the entity would only cause more issues laters.

Reply:
Added them as a debug log message, but as mentioned, I do not think adding them to the repair itself is a good idea since the list may be incomplete.

Reply:
For users who have a lot of automations and scripts and might have trouble finding which one to update, I think it is better to tell them the ones we know about since it saves them some time having to figure it out themselves.  We can also tell them its not an exhaustive list, and they should be sure to check any complex automations that may indirectly reference the sensor.